### DISCUSSION TITLE
What is the purpose of using guess bond or guess atom before calculating the fingerprint?

### DISCUSSION THREAD

**Discussion by tjrong123** [chemosim-lab/ProLIF#172] [Q&A] [answered]
From my understanding the purpose Guess Bond is to fix the topology or trajectory (not really sure) with lost information on bond connection. Can adding a modeller module to fix the topology file before passing it to MDanalysis omit the need for guess bond?

**Comment by cbouy** [comment]
ProLIF needs an input with a "proper" molecule (with all bonds defined, elements instead of atom types...etc.).
Some topology file formats don't provide this information, or provide it in an incomplete way (e.g. PDB files sometimes only have the `CONECT` bond record for small molecules but not protein atoms) so in that case you'll need to call `guess_bonds`.
If you're topology file already has all bonds explicit (in a way that MDAnalysis recognizes), then there's no need to call `guess_bonds`

**Reply by tjrong123** [reply]
Ohh i understand now,  if my topology file having PSF format I can omit the need to call guess_bond?

**Reply by cbouy** [reply, marked as answer]
Yes I think PSF always have bonds specified, although you might need to add in `elements` since PSF doesn't have them:
```python
import MDAnalysis as mda
from MDAnalysis.topology.guessers import guess_types

u =mda.Universe(PSF_file)
u.add_TopologyAttr('elements', guess_types(u.atoms.names))
```
and then you should be good to go (and guessing elements is much safer and faster than guessing bonds so it's preferable to always use a topology file format that has bonds defined)

**Reply by tjrong123** [reply]
@cbouy thanks alot.
